projects/MarlQ/proposal/README.md

```diff
@@ -0,0 +1,56 @@
+Since this kind of deformation is actually present in a lot of daily interactions (considering hands themselves can act as a sort of soft body), it makes it important for ultra-realistic animation.
+The shadows created by these deformations are important for realism.
+I imagine others would be interested in this feature as well. 
+
+## Proposal
+
+The proposed feature is a simple, momentary simulation of the volume displacement of a soft object caused by one (or more) solid object(s).
+This displacement would directly impact the vertex positions of the soft object, changing the mesh.
+Ideally, these deformations could then be saved as shape keys.
+Physical properties like elasticity or material density of the soft object could be factored in as parameters, to control the range of the volume displacement and its strength.
+However, other typical physical properties like gravity, or even push resistance should not be considered.
+The former (gravity) would create undesirable side effects. 
+The latter (push resistance) does not make sense, considering it is a momentary simulation --- this means that the hard object already intersects with the soft object which would indicate a possibly infinite force being exerted onto the soft object, rendering any push resistance meaningless.
+
+Unfortunately, I could not yet find much regarding this topic, except a video showcasing a simple method utilizing booleans for Houdini, which labels the method as 'boolean denting' (Boolean Denting (VDB Basics)).
+The results from that method look alright for some cases, but the resulting dents are way too exaggerated, making them look more artistic rather than realistic.
+Jasper Flick also made a tutorial for a similar effect for the Unity engine, mentioning the use of the inverse-square law in conjunction with the force exerted onto the soft object (Mesh Deformation - Making a Stress Ball).
+
+## Implementation
+As a Blender addon, the project will be implemented in Python.
+For now, the connected functions will be implemented one after the other, starting with a simple displacement operation, which displaces all overlapping vertices of a soft object with another, hard object.
+To enable better shadow and light interactions, here, already, can be made improvements by creating a small gap betweent the objects.
+
+Then, the remaining displaced volume will distributed across the rest of the object.
+How exactly it will be distributed remains an open question, but there multiple posibilities (e.g. simple linear falloff, inverse-square).
+Material properties could change the exact way the force caused by the volume displacement is distributed.
+Considerations will have to be made how to handle the overlapping points (when the objects intersect at multiple areas).
+
+Lastly, the dents will be smoothed, based on the materials shear resistance, as the dents usually do not form in immediate vicinity of the displacing object.
+
```

Comment:
There are Brushes in blenders sculpt mode that use cloth simulation to simulate the behaviour when one strokes over the surface or applies tension. Maybe checking out the implementation of these brushes could help you in creating this addon. Or maybe your idea could also be a brush

Reply:
Unfortunately, the cloth brush is not volume preserving, and it is always based on the brush shape, not the shape of the hard (displacing) object. I don't want to implement a brush, because I want the user to have to do as little manual work as possible.

I think I need to clarify, that the solution can't actually be a simulation, because that assumes that there is a time dimension, which there isn't for my use-case. I have to retroactively compute the most likely behaviour that the soft object should have when the hard/discplacing object already sticks into it. Because that's fast and easy for the user, and decently solvable in 99% of normal cases.

Reply:
A correct deformation direction might not be solvable. An alternative might allow user to provide such an input.